gobblin-runtime/src/main/java/org/apache/gobblin/runtime/TaskStateCollectorService.java

```diff
@@ -91,21 +103,26 @@ public TaskStateCollectorService(Properties jobProps, JobState jobState, EventBu
         Integer.parseInt(jobProps.getProperty(ConfigurationKeys.TASK_STATE_COLLECTOR_INTERVAL_SECONDS,
             Integer.toString(ConfigurationKeys.DEFAULT_TASK_STATE_COLLECTOR_INTERVAL_SECONDS)));
 
-    if (jobProps.containsKey(ConfigurationKeys.TASK_STATE_COLLECTOR_HANDLER_CLASS)){
+    // !String.equals("") checks both cases where String == null and String.isEmpty()
+    if (jobProps.containsKey(ConfigurationKeys.TASK_STATE_COLLECTOR_HANDLER_CLASS)
+        && !jobProps.getProperty(ConfigurationKeys.TASK_STATE_COLLECTOR_HANDLER_CLASS).equals("")) {
```

Comment:
Can use StringUtils.isBlank() here to support inputs like "   " which sometimes accidentally gets set in Azkaban.